packages/terra-functional-testing/src/commands/utils/describeViewports.js

```diff
@@ -0,0 +1,40 @@
+const setViewport = require('./setViewport');
+
+/**
+ * Mocha `describe` block to run tests through a defined list of viewports.
+ * This is intended to be used as an alias replacement for the root-level Mocha `describe`.
+ * Note: If a FORM_FACTOR has been specified for the test run only that form factor will be run.
+ * @param {string} title - The `describe` block title.
+ * @param {string[]} viewports - A list of viewports. [tiny, small, medium, large, huge, enormous]
+ * @param {function} - The block of tests to execute against each viewport.
+ */
+const describeViewports = (title, viewports, fn) => {
+  const { formFactor } = global.Terra.serviceOptions;
+
+  let previousViewportSize;
+
+  viewports.forEach((viewport) => {
+    // If a form factor has been specified run only that form factor.
+    if (formFactor && viewport !== formFactor) {
+      return;
+    }
+
+    global.describe(`[${viewport}]`, () => {
+      global.before(() => {
+        // Store the original browser window size so it can be restored after the test has run.
+        previousViewportSize = global.browser.getWindowSize();
+
+        setViewport(viewport);
```

Comment:
It has been mentioned before that our `describeViewports` function can be slow. I haven't done too much investigation into why, but a good place to start might be looking into how expensive accessing/resizing the window is. When we run in a specific form factor nearly all of these calls are redundant (get size, set size, reset size again). I left these in here for passivity, but I think there is some optimizations we can do here.

Reply:
I hope retrieving the window size isn't the slow part but we could add a quick check to not try to reset the viewport if the sizes are the same.